params/variants.toml

```diff
@@ -5355,25 +5355,60 @@ sampler = "ф"
 samplerExplain = "Cyrillic Lower Ef"
 tagKind = "letter"
 
-[prime.cyrl-ef.variants.serifless]
+[prime.cyrl-ef.variants-buildup]
+entry = "bowl"
+descriptionLeader = "Cyrillic Lower Ef (`ф`)"
+
+[prime.cyrl-ef.variants-buildup.stages.bowl."*"]
+next = "bar"
+
+[prime.cyrl-ef.variants-buildup.stages.bowl.standard]
 rank = 1
-description = "Cyrillic Lower Ef (`ф`) with standard shape and without serifs"
-selector."cyrl/ef"  = "serifless"
+keyAffix = ""
+selectorAffix."cyrl/ef" = ""
+selectorAffix."cyrl/ef.BGR" = ""
 
-[prime.cyrl-ef.variants.cursive]
+[prime.cyrl-ef.variants-buildup.stages.bowl.split]
 rank = 2
-description = "Cyrillic Lower Ef (`ф`) with cursive shape"
-selector."cyrl/ef"  = "cursive"
+descriptionAffix = "split bowl"
+selectorAffix."cyrl/ef" = "splitBowl"
+selectorAffix."cyrl/ef.BGR" = ""
 
-[prime.cyrl-ef.variants.top-serifed]
-rank = 3
-description = "Cyrillic Lower Ef (`ф`) with standard shape and serif at top"
-selector."cyrl/ef"  = "topSerifed"
+[prime.cyrl-ef.variants-buildup.stages.bar."*"]
+next = "serifs"
 
-[prime.cyrl-ef.variants.serifed]
-rank = 4
-description = "Cyrillic Lower Ef (`ф`) with standard shape and serifs at top and bottom"
-selector."cyrl/ef"  = "serifed"
+[prime.cyrl-ef.variants-buildup.stages.bar.straight]
+rank = 1
+keyAffix = ""
+descriptionAffix = "straight bar"
+selectorAffix."cyrl/ef" = ""
+selectorAffix."cyrl/ef.BGR" = ""
+
+[prime.cyrl-ef.variants-buildup.stages.bar.cursive]
+rank = 2
+next = "END"
+descriptionAffix = "cursive bar"
+selectorAffix."cyrl/ef" = "cursive"
+selectorAffix."cyrl/ef.BGR" = "cursive"
+
+[prime.cyrl-ef.variants-buildup.stages.serifs.serifless]
+rank = 1
+descriptionAffix = "serifs"
+descriptionJoiner = "without"
+selectorAffix."cyrl/ef" = "serifless"
+selectorAffix."cyrl/ef.BGR" = "serifless"
+
+[prime.cyrl-ef.variants-buildup.stages.serifs.top-serifed]
+rank = 2
```

Comment:
Disable for cursive.
Add `nonBreakingVariantAdditionPriority = 10` to move it to the end of variant list.